qrexec-lib/qube-name.c

```diff
@@ -0,0 +1,60 @@
+#include <stdint.h>
+#include "pure.h"
+
+QUBES_PURE_PUBLIC bool
+qubes_pure_is_valid_qube_name(const struct QubesSlice untrusted_name)
+{
+    /* Check the length first. */
+    if (untrusted_name.length < 1 || untrusted_name.length > 31)
+        return false;
+    size_t const length = untrusted_name.length;
+    uint8_t const untrusted_first = untrusted_name.pointer[0] | 0x20;
```

Comment:
`untrusted_first_lowercase` since you convert it? But also, IMO it will be clearer to simply extend the `if` below with both lowercase and uppercase, instead of doing magic on ASCII codes.